### PR TITLE
Upgrade to terraform 0.12.x

### PR DESCRIPTION
**Pull request by confusdcodr**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,80 +1,80 @@
...
 // Module: inspector
 //
-provider "aws" {}
+provider "aws" {
+}
 
 locals {
-  iam_role_specified = "${var.iam_role_arn == ""}"
-  iam_role_arn       = "${local.iam_role_specified ? join("", aws_iam_role.this.*.arn) : var.iam_role_arn}"
+  iam_role_specified = var.iam_role_arn == ""
```
Comment (userhas404d) [inline main.tf]: per the changes requested [here](https://github.com/plus3it/terraform-aws-vault/pull/1#discussion_r319090385) recommending:
```suggestion
  iam_role_specified = var.iam_role_arn == null
```

File: tests/baseline/main.tf [hunk 2]
```diff
@@ -15,11 +11,12 @@ module "inspector" {
   source = "../../"
 
   providers = {
-    aws = "aws"
+    aws = aws
   }
```
Comment (userhas404d) [inline tests/baseline/main.tf]: It might be useful to have tags assigned here that indicate that the resulting resources are part of this test.

Comment (confusdcodr) [reply tests/baseline/main.tf]: The resources are automatically deleted so you would never see the tags

File: scripts/terraform-docs.awk [hunk 3]
```diff
@@ -0,0 +1,83 @@
+# This script converts Terraform 0.12 variables/outputs to something suitable for `terraform-docs`
```
Comment (lorengordon) [inline scripts/terraform-docs.awk]: We'll save ourselves a lot of time later if we take the time now to put the tfdoc workaround in a central location

File: main.tf [hunk 4]
```diff
@@ -1,80 +1,80 @@
...
 // Module: inspector
 //
-provider "aws" {}
+provider "aws" {
+}
 
 locals {
-  iam_role_specified = "${var.iam_role_arn == ""}"
-  iam_role_arn       = "${local.iam_role_specified ? join("", aws_iam_role.this.*.arn) : var.iam_role_arn}"
+  iam_role_not_specified = var.iam_role_arn == null
```
Comment (lorengordon) [inline main.tf]:
```suggestion
  create_iam_role = var.iam_role_arn == null
```

File: scripts/terraform-docs.awk (hunk 3 above)
Comment (confusdcodr) [reply scripts/terraform-docs.awk]: Fixed

File: main.tf (hunk 4 above)
Comment (confusdcodr) [reply main.tf]: Fixed

**Pull request closed without merging**